src/queries/label-text.js

```diff
@@ -89,7 +94,13 @@ function queryAllByLabelText(
     [],
   )
 
-  return Array.from(new Set([...labelledElements, ...ariaLabelledElements]))
+  const allMatches = Array.from(
+    new Set([...labelledElements, ...ariaLabelledElements]),
+  )
+  const elementsMatchingSelector = new Set(container.querySelectorAll(selector))
+  return allMatches.filter(matchingElement =>
+    elementsMatchingSelector.has(matchingElement),
+  )
```

Comment:
I think we'd be better off with `matches` here: https://developer.mozilla.org/en-US/docs/Web/API/Element/matches

Reply:
I had a feeling this wasn't the best way. Probably could have done a little more digging haha.